crates/ibc/src/core/ics04_channel/handler.rs

```diff
@@ -290,21 +289,8 @@ fn do_packet_callback(
 
     match msg {
         PacketMsg::Recv(msg) => {
-            let result = cb.on_recv_packet(module_output, &msg.packet, &msg.signer);
-            match result {
-                OnRecvPacketAck::Nil(write_fn) => {
```

Comment:
What was the purpose of this `Nil` variant? Wouldn't be needed anymore (since we put off async ack)?

Reply:
It was meant to emulate the fact that an `Ack` in go can be `nil`, which they use to mean "this is an async acknowledgements; ibc-core please don't write an acknowledgement yet, I haven't processed the packet yet" - see [here](https://github.com/cosmos/ibc-go/blob/e92aefe04721f9fc60105c0b3341be74563c20b1/modules/core/keeper/msg_server.go#L435-L439) (there was little to no documentation about this and took me a while to figure out). Since we removed async acks for now, we don't need it anymore.

Reply:
Yes, there was little context around it. thank you